### PR TITLE
Agreement: Implement staking models

### PR DESCRIPTION
**Pull request by facuspagnuolo**
**MERGE AFTER #263**

This PR adds the missing components to support staking-related models in an agreement app:
- Staking per token<>user
- Staking movements in the context of an agreement app

### CODE REVIEW

File: packages/connect-agreement/src/thegraph/queries/index.ts [hunk 1]
```diff
@@ -120,6 +120,75 @@ export const GET_COLLATERAL_REQUIREMENT = (type: string) => gql`
...
+    }
+  }
+`
+
+export const GET_STAKING_MOVEMENTS = (type: string) => gql`
+  ${type} StakingMovements($stakingId: String!, $agreement: String!, $noAgreement: String!, $first: Int!, $skip: Int!) {
+    stakingMovements(where: {
+      staking: $stakingId,
+      agreementId_in: [$noAgreement, $agreement]
+    }, first: $first, skip: $skip) {
```
Comment (facuspagnuolo) [inline packages/connect-agreement/src/thegraph/queries/index.ts:145]: for some reason this filter is not working, maybe @0xGabi can help?

Comment (facuspagnuolo) [reply packages/connect-agreement/src/thegraph/queries/index.ts]: Fixed now!

File: packages/connect-agreement/src/thegraph/connector.ts [hunk 2]
```diff
@@ -242,6 +248,80 @@ export default class AgreementConnectorTheGraph implements IAgreementConnector {
...
+      { stakingId },
+      (result: QueryResult) => parseStaking(result, this)
+    )
+  }
+
+  onStaking(
+    stakingId: string,
+    callback: SubscriptionCallback<Staking>
+  ): SubscriptionHandler {
+    return this.#gql.subscribeToQueryWithParser<Staking>(
```
Comment (bpierre) [inline packages/connect-agreement/src/thegraph/connector.ts:265]: I think we could use `SubscriptionResult` / `subscription()` here too.

File: packages/connect-agreement/src/thegraph/connector.ts [hunk 3]
```diff
@@ -242,6 +248,80 @@ export default class AgreementConnectorTheGraph implements IAgreementConnector {
...
+    )
+  }
+
+  onStakingMovements(
+    stakingId: string,
+    agreement: string,
+    first: number,
+    skip: number,
+    callback: SubscriptionCallback<StakingMovement[]>
+  ): SubscriptionHandler {
```
Comment (bpierre) [inline packages/connect-agreement/src/thegraph/connector.ts:293]: And here too :detective:

File: packages/connect-agreement/src/thegraph/connector.ts [hunk 4]
```diff
@@ -242,6 +248,80 @@ export default class AgreementConnectorTheGraph implements IAgreementConnector {
...
+      queries.GET_ACTION('query'),
+      { actionId },
+      (result: QueryResult) => parseAction(result, this)
+    )
+  }
+
+  onAction(
+    actionId: string,
+    callback: SubscriptionCallback<Action | null>
+  ): SubscriptionHandler {
```
Comment (bpierre) [inline packages/connect-agreement/src/thegraph/connector.ts:316]: …and here :smile:

**Review by bpierre — APPROVED** [review]

**Pull request merged**